lsp/nls/src/usage.rs

```diff
@@ -162,6 +162,34 @@ impl UsageLookup {
 
                         TraverseControl::ContinueWithScope(new_env)
                     }
+                    Term::App(f, value) => {
```

Comment:
Should we also do this even when the match isn't applied, like a function definition `let foo = match { 'Bar payload => p[.] }` to get `payload` completion on `p` (and maybe even record completion if we have type information on the argument of `foo`)?

Reply:
Good point, I've added that. I'm having some trouble coming up with a realistic test that uses type information to get completions. The problem is that if typechecking fails then completions just don't work at all right now. (I'll open an issue for this.) This means that something like
```
fun x => match { y => y.fo } : { foo: Number} -> Number
```
fails to type-check and therefore doesn't offer completions. I've made it "work" for now by writing
```
fun x => match { y => y.fo } : { foo: Number, fo: Number} -> Number
```